File: powervoice/src/main/java/com/example/powervoice/model/Report.java
package com.example.powervoice.model;

import java.io.Serializable;
import java.sql.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;

import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Entity
public class Report implements Serializable{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, updatable = false)
    private Long id;
    
    @ManyToOne(optional = false)
    private ReportCategory reportCategory;

    @Column(nullable = false)
    private Date reportDate;

    @ManyToOne(optional = false)
    private User user;    
    
    @ManyToOne(optional = false)
    private Feature feature;

    @Column
    private String description;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private EReportStatus reportStatus;

}

File: powervoice-frontend/src/app/models/reports/report.ts
import { Data } from "@angular/router";
import { Feature } from "../features/feature.model";
import { ReportCategory } from "../report-category/report-category";
import { User } from "../users/user";


export interface Report {
    id : number;
    description : string;
    reportDate : Date;
    reportStatus : string; 
    feature : Feature;
    reportCategory: ReportCategory; 
    user : User; 

}
